File: BC-W4/Army/Spells/LesserHeal/LesserHeal.cpp
#include "LesserHeal.h"

LesserHeal::LesserHeal(int actionPoints, int manaCost, const std::string& spellName) : Spell(actionPoints, manaCost, spellName) {};
LesserHeal::~LesserHeal() {};

void LesserHeal::action(Unit* enemy) {
    throw InvalidActionException();
};

void LesserHeal::heal(Unit* unit) {
    unit->addHitPoints(this->getActionPoints());
};

File: BC-W4/ArmyJava/src/spells/LesserHeal.java
package spells;

import exceptions.InvalidCastException;
import units.Unit;

public class LesserHeal extends Spell {
    public LesserHeal() {
        super("LesserHeal", 30, 40, false);
    }

    public LesserHeal(int actionPoints) {
        super("LesserHeal", 30, actionPoints, false);
    }

    public void action(Unit enemy) throws InvalidCastException {
        throw new InvalidCastException();
    }

    public void heal(Unit unit) throws InvalidCastException {
        unit.addHitPoints(this.actionPoints);
    }
}
